### DISCUSSION TITLE
Run core-common-config-bootstrapper without configuration provider

### DISCUSSION THREAD

**Discussion by lindseysimple** [edgexfoundry/community#131] [Q&A] [answered]
Hi all,

I followed the instructions of [install-and-deploy-native common-config](https://github.com/edgexfoundry/edgex-go/tree/main/internal/core/common_config#install-and-deploy-native) and would like to execute the `core-common-config-bootstrapper` binary executable locally in my dev environment without using the config provider.
```
# run the microservice (may require other dependent services to run correctly) and provide the configuration provider using -cp
./core-common-config-bootstrapper
```

It returned the following error:
```
level=INFO ts=2023-06-19T10:16:03.437812Z app=core-common-config-bootstrapper source=main.go:63 msg="Core Common Config is starting"
level=INFO ts=2023-06-19T10:16:03.438183Z app=core-common-config-bootstrapper source=main.go:75 msg="Secret Provider created"
level=INFO ts=2023-06-19T10:16:03.438212Z app=core-common-config-bootstrapper source=main.go:84 msg="Got Config Provider Access Token with length 0"
level=INFO ts=2023-06-19T10:16:03.438228Z app=core-common-config-bootstrapper source=config.go:629 msg="Using Configuration provider () from: http://:0 with base path of edgex/v3/core-common-config-bootstrapper"
level=ERROR ts=2023-06-19T10:16:03.438244Z app=core-common-config-bootstrapper source=main.go:97 msg="failed to create provider client for the common configuration: unable to create Configuration Client: Configuration service host and/or port or serviceKey not set"
```
From the help text it seems that using the `-cc` option might work.
```
Usage: ./core-common-config-bootstrapper [options]
Server Options:
    -cp, --configProvider           Indicates to use Configuration Provider service at specified URL.
                                    URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500
    -cc, --commonConfig             Takes the location where the common configuration is loaded from when
                                    not using the Configuration Provider
    -o, --overwrite                 Overwrite configuration in provider with local configuration
                                    *** Use with cation *** Use will clobber existing settings in provider,
                                    problematic if those settings were edited by hand intentionally
    -cf, --configFile <name>        Indicates name of the local configuration file. Defaults to configuration.toml
    -p, --profile <name>            Indicate configuration profile other than default
    -cd, --configDir                Specify local configuration directory
    -r, --registry                  Indicates service should use Registry.
    -d, --dev                       Indicates service to run in developer mode which causes Host configuration values to be overridden.
                                    with `localhost`. This is so that it will ru
```

Tried to use `./core-common-config-bootstrapper -cc=./res` but got the same result.

Is there any instructions to run the `core-common-config-bootstrapper` executable without configuration provider?
Thanks for your suggestion.

**Comment by farshidtz** [comment, marked as answer]
Hi,

The sole responsibility of `core-common-config-bootstrapper` is to take the common configuration from a file and push them into the Config Provider. Without the Config Provider, `core-common-config-bootstrapper` serves no purpose.

Instead, you would need to set the `-cc, --commonConfig` of other services to point to the common configuration file, or have those common configuration bits added to local configuration files.

**Reply by lindseysimple** [reply]
Thanks @farshidtz for the detailed explanation , that makes sense to me!

**Reply by lenny-goodell** [reply]
@lindseysimple , also see here for more details. https://docs.edgexfoundry.org/3.0/microservices/configuration/CommonConfiguration/
You can copy the file here https://github.com/edgexfoundry/edgex-go/blob/v3.0/cmd/core-common-config-bootstrapper/res/configuration.yaml and set it with  `-cc/--commonConfig`  on each service.
